v2/api/documentdb/customizations/sql_database_extension_types.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package customizations
+
+import (
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/go-logr/logr"
+
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/core"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+)
+
+var _ extensions.ErrorClassifier = &SqlDatabaseExtension{}
+
+// ClassifyError evaluates the provided error, returning whether it is fatal or can be retried.
+// A BadRequest (400) is normally fatal, but CosmosDB Databases may return 400 if database creation is attempted while
+// the parent account is still being created, so we make BadRequest retryable for this case.
+// cloudError is the error returned from ARM.
+// apiVersion is the ARM API version used for the request.
+// log is a logger than can be used for telemetry.
+// next is the next implementation to call.
+func (e *SqlDatabaseExtension) ClassifyError(
+	cloudError *genericarmclient.CloudError,
+	apiVersion string,
+	log logr.Logger,
+	next extensions.ErrorClassifierFunc) (core.CloudErrorDetails, error) {
+	details, err := next(cloudError)
+	if err != nil {
+		return core.CloudErrorDetails{}, err
+	}
+
+	// Override is to treat BadRequest as retryable for SqlDatabases
+	if isRetryableBadRequest(cloudError.InnerError) {
+		details.Classification = core.ErrorRetryable
+	}
+
+	return details, nil
+}
+
+// isRetryableConflict checks the passed error to see if it is a retryable conflict, returning true if it is.
+func isRetryableBadRequest(err *genericarmclient.ErrorResponse) bool {
+	if err == nil || err.Message == nil {
```

Comment:
A bit weird you check `err.Message` here but then use the `Code` below? What if they sent a `Code` but no `Message`?

I think just the `Code` check below is sufficient and you could remove this `Message == nil` bit.

Reply:
Agreed.